src/Entity/ProductBundle.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Drupal\commerce_product\Entity\ProductBundle;
+
+use Drupal\commerce\PurchasableEntityInterface;
+use Drupal\commerce_product\Entity\Product;
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+
+/**
+ * Defines the product entity class.
+ *
+ * @ContentEntityType(
+ *   id = "commerce_product_bundle",
+ *   label = @Translation("Product Bundle"),
+ *   label_singular = @Translation("product bundle"),
+ *   label_plural = @Translation("product bundles"),
+ *   label_count = @PluralTranslation(
+ *     singular = "@count product bundle",
+ *     plural = "@count product bundles",
+ *   ),
+ *   handlers = {
+ *     "event" = "Drupal\commerce_product\Event\ProductEvent",
+ *     "storage" = "Drupal\commerce\CommerceContentEntityStorage",
+ *     "access" = "\Drupal\commerce\EntityAccessControlHandler",
+ *     "permission_provider" = "\Drupal\commerce\EntityPermissionProvider",
+ *     "view_builder" = "Drupal\Core\Entity\EntityViewBuilder",
+ *     "list_builder" = "Drupal\commerce_product\ProductListBuilder",
+ *     "views_data" = "Drupal\views\EntityViewsData",
+ *     "form" = {
+ *       "default" = "Drupal\commerce_product\Form\ProductForm",
+ *       "add" = "Drupal\commerce_product\Form\ProductForm",
+ *       "edit" = "Drupal\commerce_product\Form\ProductForm",
+ *       "delete" = "Drupal\Core\Entity\ContentEntityDeleteForm"
+ *     },
+ *     "route_provider" = {
+ *       "default" = "Drupal\Core\Entity\Routing\AdminHtmlRouteProvider",
+ *       "delete-multiple" = "Drupal\entity\Routing\DeleteMultipleRouteProvider",
+ *     },
+ *     "translation" = "Drupal\commerce_product\ProductTranslationHandler"
+ *   },
+ *   admin_permission = "administer commerce_product_bundle",
+ *   permission_granularity = "bundle",
+ *   fieldable = TRUE,
+ *   translatable = TRUE,
+ *   base_table = "commerce_product_bundle",
+ *   data_table = "commerce_product_bundle_data",
+ *   entity_keys = {
+ *     "id" = "product_bundle_id",
+ *     "bundle" = "type",
+ *     "label" = "title",
+ *     "langcode" = "langcode",
+ *     "uuid" = "uuid",
+ *     "status" = "status",
+ *   },
+ *   links = {
+ *     "canonical" = "/product/{commerce_product}",
+ *     "add-page" = "/product/add",
+ *     "add-form" = "/product/add/{commerce_product_type}",
+ *     "edit-form" = "/product/{commerce_product}/edit",
+ *     "delete-form" = "/product/{commerce_product}/delete",
+ *     "delete-multiple-form" = "/admin/commerce/products/delete",
+ *     "collection" = "/admin/commerce/products"
+ *   },
+ *   bundle_entity_type = "commerce_product_type",
```

Comment:
i think this will be commerce_product_bundle_static -- the point is I don't think we need 'bundles' or variations of this bundle type.